drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -63,6 +63,19 @@ static struct nrf5_802154_data nrf5_data;
 #define FRAME_PENDING_BIT (1 << 4)
 #define TXTIME_OFFSET_US  (5 * USEC_PER_MSEC)
 
+#if defined(CONFIG_IEEE802154_UICR_EUI64_ENABLE)
+#if defined(CONFIG_SOC_NRF5340_CPUAPP) || defined(CONFIG_SOC_NRF5340_CPUNET)
+#if defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
+#define EUI64_ADDR (NRF_UICR_S->OTP)
+#else
+#define EUI64_ADDR (NRF_UICR->OTP)
+#endif
+#else
+#define EUI64_ADDR (NRF_UICR->CUSTOMER)
+#endif
+#define EUI64_ADDR_HIGH CONFIG_IEEE802154_UICR_EUI64_REG
+#define EUI64_ADDR_LOW (CONFIG_IEEE802154_UICR_EUI64_REG + 1)
+#else
```

Comment:
Could you add comments to each `#else` and `#endif`? It's very hard to follow now

Reply:
Indeed.